File: MrCream.API/Controller/YoghurtController.cs
using Microsoft.AspNetCore.Mvc;

namespace MrCream.API.Controllers;

[ApiController]
[Route("api/[controller]")]
public class YoghurtController : ControllerBase
{
    private readonly ILogger<YoghurtController> _logger;

    public YoghurtController(ILogger<YoghurtController> logger)
    {
        _logger = logger;
    }

    /// <summary>
    /// Get all available yoghurt flavours
    /// </summary>
    [HttpGet("flavours")]
    public ActionResult<IEnumerable<YoghurtFlavour>> GetFlavours()
    {
        var flavours = new List<YoghurtFlavour>
        {
            new YoghurtFlavour
            {
                Id = 1,
                Name = "MrCream Strawberry",
                Description = "Sweet, creamy, and timeless. Our strawberry yoghurt drink is a classic favorite, with just the right blend of tang and berry-rich flavor.",
                Price = 300.00m,
                Size = "500ml",
                ImageUrl = "/images/yoghurt/strawberry.jpg",
                IsAvailable = true,
                NutritionalInfo = new NutritionalInfo
                {
                    Calories = 180,
                    Protein = "8g",
                    Calcium = "25% DV",
                    Probiotics = "Live cultures"
                }
            },
            new YoghurtFlavour
            {
                Id = 2,
                Name = "MrCream Vanilla",
                Description = "Mellow, smooth, and aromatic. Vanilla lovers will enjoy this elegant twist on creamy indulgence — perfect chilled!",
                Price = 300.00m,
                Size = "500ml",
                ImageUrl = "/images/yoghurt/vanilla.jpg",
                IsAvailable = true,
                NutritionalInfo = new NutritionalInfo
                {
                    Calories = 170,
                    Protein = "8g",
                    Calcium = "25% DV",
                    Probiotics = "Live cultures"
                }
            },
            new YoghurtFlavour
            {
                Id = 3,
                Name = "MrCream Greek Style",
                Description = "Coming Soon - Thick, rich, and protein-packed Greek yoghurt experience.",
                Price = 350.00m,
                Size = "400ml",
                ImageUrl = "/images/yoghurt/greek.jpg",
                IsAvailable = false,
                NutritionalInfo = new NutritionalInfo
                {
                    Calories = 150,
                    Protein = "15g",
                    Calcium = "30% DV",
                    Probiotics = "Live cultures"
                }
            }
        };

        return Ok(flavours);
    }

    /// <summary>
    /// Get specific yoghurt flavour by ID
    /// </summary>
    [HttpGet("flavours/{id}")]
    public ActionResult<YoghurtFlavour> GetFlavour(int id)
    {
        // This would typically come from database
        var flavour = GetFlavours().Value?.FirstOrDefault(f => f.Id == id);

        if (flavour == null)
        {
            return NotFound($"Yoghurt flavour with ID {id} not found.");
        }

        return Ok(flavour);
    }

    /// <summary>
    /// Place an order for yoghurt drinks
    /// </summary>
    [HttpPost("order")]
    public ActionResult<OrderResponse> PlaceOrder([FromBody] YoghurtOrderRequest request)
    {
        if (request == null || !request.Items.Any())
        {
            return BadRequest("Order must contain at least one item.");
        }

        // Calculate total
        var flavours = GetFlavours().Value?.ToList() ?? new List<YoghurtFlavour>();
        decimal total = 0;
        var orderItems = new List<OrderItem>();

        foreach (var item in request.Items)
        {
            var flavour = flavours.FirstOrDefault(f => f.Id == item.FlavourId);
            if (flavour == null)
            {
                return BadRequest($"Flavour with ID {item.FlavourId} not found.");
            }

            if (!flavour.IsAvailable)
            {
                return BadRequest($"{flavour.Name} is currently not available.");
            }

            var itemTotal = flavour.Price * item.Quantity;
            total += itemTotal;

            orderItems.Add(new OrderItem
            {
                FlavourId = item.FlavourId,
                FlavourName = flavour.Name,
                Quantity = item.Quantity,
                UnitPrice = flavour.Price,
                SubTotal = itemTotal
            });
        }

        var response = new OrderResponse
        {
            OrderId = Guid.NewGuid().ToString(),
            Items = orderItems,
            SubTotal = total,
            DeliveryFee = request.DeliveryAddress != null ? 500.00m : 0m,
            Total = total + (request.DeliveryAddress != null ? 500.00m : 0m),
            CustomerInfo = request.CustomerInfo,
            DeliveryAddress = request.DeliveryAddress,
            OrderDate = DateTime.UtcNow,
            EstimatedDelivery = request.DeliveryAddress != null
                ? DateTime.UtcNow.AddHours(2)
                : DateTime.UtcNow.AddMinutes(30),
            Status = "Confirmed",
            PaymentMethod = request.PaymentMethod
        };

        _logger.LogInformation("Yoghurt order placed: {OrderId} for {Total:C}", response.OrderId, response.Total);

        return Ok(response);
    }

    /// <summary>
    /// Get nutritional information for all flavours
    /// </summary>
    [HttpGet("nutrition")]
    public ActionResult<IEnumerable<object>> GetNutritionalInfo()
    {
        var nutritionInfo = GetFlavours().Value?.Select(f => new
        {
            f.Id,
            f.Name,
            Nutrition = f.NutritionalInfo
        });

        return Ok(nutritionInfo);
    }

    /// <summary>
    /// Get store locations where yoghurt is available
    /// </summary>
    [HttpGet("stores")]
    public ActionResult<IEnumerable<StoreLocation>> GetStoreLocations()
    {
        var stores = new List<StoreLocation>
        {
            new StoreLocation
            {
                Id = 1,
                Name = "MrCream Head Office Store",
                Address = "18-26 Dauda Ayorinde Street Adaloko, Afromedia, off Badagry Expressway, Lagos State",
                Phone = "+234 803 235 4952",
                IsOpen = true,
                OpeningHours = "Mon-Fri: 8AM-6PM, Sat-Sun: 9AM-5PM"
            },
            new StoreLocation
            {
                Id = 2,
                Name = "Shoprite - Various Locations",
                Address = "Multiple locations across Lagos",
                Phone = "Contact individual stores",
                IsOpen = true,
                OpeningHours = "Daily: 8AM-10PM"
            }
        };

        return Ok(stores);
    }
}

// Data Models for Yoghurt API
public class YoghurtFlavour
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string Size { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public bool IsAvailable { get; set; }
    public NutritionalInfo NutritionalInfo { get; set; } = new();
}

public class NutritionalInfo
{
    public int Calories { get; set; }
    public string Protein { get; set; } = string.Empty;
    public string Calcium { get; set; } = string.Empty;
    public string Probiotics { get; set; } = string.Empty;
}

public class YoghurtOrderRequest
{
    public List<OrderItemRequest> Items { get; set; } = new();
    public CustomerInfo CustomerInfo { get; set; } = new();
    public DeliveryAddress? DeliveryAddress { get; set; }
    public string PaymentMethod { get; set; } = "Cash"; // Cash, Card, Transfer
}

public class OrderItemRequest
{
    public int FlavourId { get; set; }
    public int Quantity { get; set; }
}

public class CustomerInfo
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
}

public class DeliveryAddress
{
    public string Street { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
}

public class OrderResponse
{
    public string OrderId { get; set; } = string.Empty;
    public List<OrderItem> Items { get; set; } = new();
    public decimal SubTotal { get; set; }
    public decimal DeliveryFee { get; set; }
    public decimal Total { get; set; }
    public CustomerInfo CustomerInfo { get; set; } = new();
    public DeliveryAddress? DeliveryAddress { get; set; }
    public DateTime OrderDate { get; set; }
    public DateTime EstimatedDelivery { get; set; }
    public string Status { get; set; } = string.Empty;
    public string PaymentMethod { get; set; } = string.Empty;
}

public class OrderItem
{
    public int FlavourId { get; set; }
    public string FlavourName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal SubTotal { get; set; }
}

public class StoreLocation
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public bool IsOpen { get; set; }
    public string OpeningHours { get; set; } = string.Empty;
}

File: api-php/controllers/YoghurtController.php
<?php
/**
 * Yoghurt Controller
 * Handles all yoghurt product related endpoints
 */

class YoghurtController {

    /**
     * GET /api/yoghurt/flavours
     * Get all available yoghurt flavours
     */
    public function getFlavours($id = null) {
        if ($id !== null) {
            // Handle /api/yoghurt/flavours/{id}
            return $this->getFlavour($id);
        }

        $flavours = [
            [
                'id' => 1,
                'Id' => 1, // Keep for backward compatibility
                'name' => 'Strawberry',
                'fullName' => 'MrCream Strawberry Flavoured Sweetened Yoghurt',
                'Name' => 'MrCream Strawberry', // Keep for backward compatibility
                'description' => 'Sweet, creamy, and timeless. Our strawberry yoghurt drink is a classic favorite, with just the right blend of tang and berry-rich flavor.',
                'Description' => 'Sweet, creamy, and timeless. Our strawberry yoghurt drink is a classic favorite, with just the right blend of tang and berry-rich flavor.',
                'Price' => 300.00,
                'Size' => '500ml',
                'ImageUrl' => '/images/yoghurt/strawberry.jpg',
                'IsAvailable' => true,
                'NutritionalInfo' => [
                    'Calories' => 180,
                    'Protein' => '8g',
                    'Calcium' => '25% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ],
            [
                'id' => 2,
                'Id' => 2,
                'name' => 'Vanilla',
                'fullName' => 'MrCream Vanilla Flavoured Sweetened Yoghurt',
                'Name' => 'MrCream Vanilla',
                'description' => 'Mellow, smooth, and aromatic. Vanilla lovers will enjoy this elegant twist on creamy indulgence — perfect chilled!',
                'Description' => 'Mellow, smooth, and aromatic. Vanilla lovers will enjoy this elegant twist on creamy indulgence — perfect chilled!',
                'Price' => 300.00,
                'Size' => '500ml',
                'ImageUrl' => '/images/yoghurt/vanilla.jpg',
                'IsAvailable' => true,
                'NutritionalInfo' => [
                    'Calories' => 170,
                    'Protein' => '8g',
                    'Calcium' => '25% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ],
            [
                'id' => 3,
                'Id' => 3,
                'name' => 'Greek Style',
                'fullName' => 'MrCream Greek Style Yoghurt',
                'Name' => 'MrCream Greek Style',
                'description' => 'Coming Soon - Thick, rich, and protein-packed Greek yoghurt experience.',
                'Description' => 'Coming Soon - Thick, rich, and protein-packed Greek yoghurt experience.',
                'Price' => 350.00,
                'Size' => '400ml',
                'ImageUrl' => '/images/yoghurt/greek.jpg',
                'IsAvailable' => false,
                'NutritionalInfo' => [
                    'Calories' => 150,
                    'Protein' => '15g',
                    'Calcium' => '30% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ]
        ];

        sendJson($flavours);
    }

    /**
     * GET /api/yoghurt/flavours/{id}
     * Get specific yoghurt flavour by ID
     */
    public function getFlavour($id) {
        $flavours = [
            1 => [
                'id' => 1,
                'Id' => 1,
                'name' => 'Strawberry',
                'fullName' => 'MrCream Strawberry Flavoured Sweetened Yoghurt',
                'Name' => 'MrCream Strawberry',
                'description' => 'Sweet, creamy, and timeless. Our strawberry yoghurt drink is a classic favorite, with just the right blend of tang and berry-rich flavor.',
                'Description' => 'Sweet, creamy, and timeless. Our strawberry yoghurt drink is a classic favorite, with just the right blend of tang and berry-rich flavor.',
                'Price' => 300.00,
                'Size' => '500ml',
                'ImageUrl' => '/images/yoghurt/strawberry.jpg',
                'IsAvailable' => true,
                'NutritionalInfo' => [
                    'Calories' => 180,
                    'Protein' => '8g',
                    'Calcium' => '25% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ],
            2 => [
                'id' => 2,
                'Id' => 2,
                'name' => 'Vanilla',
                'fullName' => 'MrCream Vanilla Flavoured Sweetened Yoghurt',
                'Name' => 'MrCream Vanilla',
                'description' => 'Mellow, smooth, and aromatic. Vanilla lovers will enjoy this elegant twist on creamy indulgence — perfect chilled!',
                'Description' => 'Mellow, smooth, and aromatic. Vanilla lovers will enjoy this elegant twist on creamy indulgence — perfect chilled!',
                'Price' => 300.00,
                'Size' => '500ml',
                'ImageUrl' => '/images/yoghurt/vanilla.jpg',
                'IsAvailable' => true,
                'NutritionalInfo' => [
                    'Calories' => 170,
                    'Protein' => '8g',
                    'Calcium' => '25% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ],
            3 => [
                'id' => 3,
                'Id' => 3,
                'name' => 'Greek Style',
                'fullName' => 'MrCream Greek Style Yoghurt',
                'Name' => 'MrCream Greek Style',
                'description' => 'Coming Soon - Thick, rich, and protein-packed Greek yoghurt experience.',
                'Description' => 'Coming Soon - Thick, rich, and protein-packed Greek yoghurt experience.',
                'Price' => 350.00,
                'Size' => '400ml',
                'ImageUrl' => '/images/yoghurt/greek.jpg',
                'IsAvailable' => false,
                'NutritionalInfo' => [
                    'Calories' => 150,
                    'Protein' => '15g',
                    'Calcium' => '30% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ]
        ];

        if (!isset($flavours[$id])) {
            sendError("Yoghurt flavour with ID $id not found.", 404);
        }

        sendJson($flavours[$id]);
    }

    /**
     * POST /api/yoghurt/order
     * Place an order for yoghurt drinks
     */
    public function postOrder() {
        $request = getJsonInput();

        if (!isset($request['Items']) || empty($request['Items'])) {
            sendError('Order must contain at least one item.', 400);
        }

        $flavours = [
            1 => ['Id' => 1, 'id' => 1, 'Name' => 'MrCream Strawberry', 'name' => 'Strawberry', 'Price' => 300.00, 'IsAvailable' => true],
            2 => ['Id' => 2, 'id' => 2, 'Name' => 'MrCream Vanilla', 'name' => 'Vanilla', 'Price' => 300.00, 'IsAvailable' => true],
            3 => ['Id' => 3, 'id' => 3, 'Name' => 'MrCream Greek Style', 'name' => 'Greek Style', 'Price' => 350.00, 'IsAvailable' => false]
        ];

        $orderItems = [];
        $total = 0;

        foreach ($request['Items'] as $item) {
            $flavourId = $item['FlavourId'];

            if (!isset($flavours[$flavourId])) {
                sendError("Flavour with ID $flavourId not found.", 400);
            }

            $flavour = $flavours[$flavourId];

            if (!$flavour['IsAvailable']) {
                sendError("{$flavour['Name']} is currently not available.", 400);
            }

            $quantity = $item['Quantity'];
            $itemTotal = $flavour['Price'] * $quantity;
            $total += $itemTotal;

            $orderItems[] = [
                'FlavourId' => $flavourId,
                'FlavourName' => $flavour['Name'],
                'Quantity' => $quantity,
                'UnitPrice' => $flavour['Price'],
                'SubTotal' => $itemTotal
            ];
        }

        $hasDeliveryAddress = isset($request['DeliveryAddress']) && !empty($request['DeliveryAddress']);
        $deliveryFee = $hasDeliveryAddress ? 500.00 : 0.00;

        $response = [
            'OrderId' => uniqid('order_'),
            'Items' => $orderItems,
            'SubTotal' => $total,
            'DeliveryFee' => $deliveryFee,
            'Total' => $total + $deliveryFee,
            'CustomerInfo' => $request['CustomerInfo'] ?? [],
            'DeliveryAddress' => $request['DeliveryAddress'] ?? null,
            'OrderDate' => date('c'),
            'EstimatedDelivery' => $hasDeliveryAddress ? date('c', strtotime('+2 hours')) : date('c', strtotime('+30 minutes')),
            'Status' => 'Confirmed',
            'PaymentMethod' => $request['PaymentMethod'] ?? 'Cash'
        ];

        logMessage("Yoghurt order placed: {$response['OrderId']} for ₦{$response['Total']}");

        sendJson($response);
    }

    /**
     * GET /api/yoghurt/nutrition
     * Get nutritional information for all flavours
     */
    public function getNutrition() {
        $nutrition = [
            [
                'Id' => 1,
                'Name' => 'MrCream Strawberry',
                'Nutrition' => [
                    'Calories' => 180,
                    'Protein' => '8g',
                    'Calcium' => '25% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ],
            [
                'Id' => 2,
                'Name' => 'MrCream Vanilla',
                'Nutrition' => [
                    'Calories' => 170,
                    'Protein' => '8g',
                    'Calcium' => '25% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ],
            [
                'Id' => 3,
                'Name' => 'MrCream Greek Style',
                'Nutrition' => [
                    'Calories' => 150,
                    'Protein' => '15g',
                    'Calcium' => '30% DV',
                    'Probiotics' => 'Live cultures'
                ]
            ]
        ];

        sendJson($nutrition);
    }

    /**
     * GET /api/yoghurt/stores
     * Get store locations where yoghurt is available
     */
    public function getStores() {
        $stores = [
            [
                'Id' => 1,
                'Name' => 'MrCream Head Office Store',
                'Address' => '18-26 Dauda Ayorinde Street Adaloko, Afromedia, off Badagry Expressway, Lagos State',
                'Phone' => '+234 803 235 4952',
                'IsOpen' => true,
                'OpeningHours' => 'Mon-Fri: 8AM-6PM, Sat-Sun: 9AM-5PM'
            ],
            [
                'Id' => 2,
                'Name' => 'Shoprite - Various Locations',
                'Address' => 'Multiple locations across Lagos',
                'Phone' => 'Contact individual stores',
                'IsOpen' => true,
                'OpeningHours' => 'Daily: 8AM-10PM'
            ]
        ];

        sendJson($stores);
    }
}
